SimulationAgent/Runtime/Config.cs

```diff
@@ -56,12 +56,12 @@ public Config(IConfigData configData)
 
             var limitsConf = new RateLimitingConfiguration
             {
-                ConnectionsPerSecond = configData.GetDouble(CONNECTIONS_FREQUENCY_LIMIT_KEY),
-                RegistryOperationsPerMinute = configData.GetDouble(REGISTRYOPS_FREQUENCY_LIMIT_KEY),
-                DeviceMessagesPerSecond = configData.GetDouble(DEVICE_MESSAGES_FREQUENCY_LIMIT_KEY),
-                DeviceMessagesPerDay = configData.GetDouble(DEVICE_MESSAGES_DAILY_LIMIT_KEY),
-                TwinReadsPerSecond = configData.GetDouble(TWIN_READS_FREQUENCY_LIMIT_KEY),
-                TwinWritesPerSecond = configData.GetDouble(TWIN_WRITES_FREQUENCY_LIMIT_KEY)
+                ConnectionsPerSecond = configData.GetInt(CONNECTIONS_FREQUENCY_LIMIT_KEY, 50),
```

Comment:
Why did you decide to go int?  I thought you were pretty set on decimals being part of the featureset?

Reply:
the RatedCounter logic needs to check for equality ("did the counter reach threshold N?") which can fail with real numbers (e.g. 2 == 2.0000000001), so rather than worrying about that, and considering that I had to rewrite the RatedCounter logic to be thread safe, I went for the simpler approach for the time being (see also notes about performance in RatedCounter), which does the job for PP